src/ucp/core/ucp_am.c

```diff
@@ -1153,6 +1178,13 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_am_handler_reply,
 
     reply_ep = ucp_worker_get_ep_by_id(worker, hdr->ep_id);
 
+    if (ucs_unlikely(!ucp_am_ep_is_valid(reply_ep))) {
+        /* ep is closed or failed, drop the data */
+        ucs_trace_data("worker %p: drop AM (reply proto) on closed/failed ep %p",
+                       worker, reply_ep);
```

Comment:
maybe add common macro to resolve ep from ep_id and debug+return if not found
ep = UCP_WORKER_GET_EP_BY_ID(worker, _ep_id, "AM with reply", return UCS_OK)
or
UCP_WORKER_GET_EP_BY_ID(worker, _ep_id, "AM with reply", return UCS_OK, &ep)

Reply:
added